src/components/DatePicker/DatePicker.jsx

```diff
@@ -182,32 +204,67 @@ export const DatePicker = ({
                     </div>
                 ) : null}
 
-                <DayPicker
-                    className={clsx(
-                        "ui-date-picker rounded-lg pt-3",
-                        useDateRangeStyle ? "date-range-picker" : null,
-                        getDayContent ? "has-custom-content" : null,
-                        modifiers.waitlist ? "has-custom-content" : null,
-                    )}
-                    todayButton={variant === "single" ? "Today" : undefined}
-                    selectedDays={selectedDays}
-                    month={currentMonth}
-                    modifiers={{ ...modifiers, ...rangeModifier }}
-                    numberOfMonths={isRangeVariant ? 2 : 1}
-                    disabledDays={disabledDays}
-                    captionElement={captionElement}
-                    renderDay={renderDay}
-                    navbarElement={NavbarElement}
-                    onDayClick={handleDayClick}
-                    onMonthChange={handleMonthChange}
-                    onTodayButtonClick={handleTodayClick}
-                    {...rest}
-                />
+                {isRangeVariant ? (
```

Comment:
Can you extract some code around this to a separate component. This component is getting really big. There are three different invocations of the the `CustomDayPicker` based on different arguments

Reply:
Also feel free to extract it into a new file to make things cleaner if needed.